lib/schema.js

```diff
@@ -473,9 +473,39 @@ Schema.prototype.defaultOptions = function(options) {
   return options;
 };
 
+/**
+ * Inherit a Schema by applying a discriminator on an existing Schema.
+ *
+ *
+ * ####Example:
+ *
+ *     const options = { discriminatorKey: 'kind' };
+ *
+ *     const eventSchema = new mongoose.Schema({ time: Date }, options);
+ *     const Event = mongoose.model('Event', eventSchema);
+ *
+ *     // ClickedLinkEvent is a special type of Event that has
+ *     // a URL.
+ *     const ClickedLinkEvent = Event.discriminator('ClickedLink',
+ *     new mongoose.Schema({ url: String }, options));
+ *
+ *     // When you create a generic event, it can't have a URL field...
+ *     const genericEvent = new Event({ time: Date.now(), url: 'google.com' });
+ *     assert.ok(!genericEvent.url);
+ *     // But a ClickedLinkEvent can
+ *     const clickedEvent = new ClickedLinkEvent({ time: Date.now(), url: 'google.com' });
+ *     assert.ok(clickedEvent.url);
+ *
+ * @param {String} name the name of the discriminator
+ * @param {Schema} schema the Schema of the discriminated Schema
+ * @return {Schema} the Schema instance
+ * @api public
+ */
+
 Schema.prototype.discriminator = function(name, schema) {
-  this._applyDiscriminators = {};
-  this._applyDiscriminators[name] = schema;
+  this._applyDiscriminators = Object.assign({}, this._applyDiscriminators, { [name]: schema });
```

Comment:
There's no need to create a new object every single time. Just `Object.assign(this._applyDiscriminators, { [name]: schema });` should be enough.

Reply:
Unit Tests fail if i apply your Suggestion